src/configuration/configuration.repository.ts

```diff
@@ -0,0 +1,81 @@
+import { ServiceConfiguration } from './entities/service-configuration.entity';
+/**
+ * Repository class for storing services and configurations in memory.
+ */
+export class ServiceConfigurationRepository {
```

Comment:
Because the configuration service is injected into other services (-> singleton) and it requires one ServiceConfigurationRepository.
So its handled "upstream"